src/commands/appdistribution-distribute.ts

```diff
@@ -0,0 +1,134 @@
+  }
+}
+
+function getAppId(appId: string): string {
+  if (!appId) {
+    throw new FirebaseError("set the --app option to a valid Firebase app id and try again");
+  }
+  return appId;
+}
+
+function getDistribution(distributionFile: string): Distribution {
+  if (!distributionFile) {
+    throw new FirebaseError("must specify a distribution file");
+  }
+
+  const distributionType = distributionFile.split(".").pop();
+  if (
+    distributionType !== DistributionFileType.IPA &&
+    distributionType !== DistributionFileType.APK
+  ) {
+    throw new FirebaseError("unsupported distribution file format, should be .ipa or .apk");
+  }
+
+  ensureFileExists(distributionFile, "verify that file points to a distribution");
+  return new Distribution(distributionFile, distributionType);
+}
+
+function getReleaseNotes(releaseNotes: string, releaseNotesFile: string): string {
+  if (releaseNotes) {
+    return releaseNotes;
+  } else if (releaseNotesFile) {
+    ensureFileExists(releaseNotesFile);
+    return fs.readFileSync(releaseNotesFile, "utf8");
+  } else {
+    return "";
+  }
+}
+
+function getTestersOrGroups(value: string, file: string): string[] {
+  if (!value && file) {
+    ensureFileExists(file);
+    value = fs.readFileSync(file, "utf8");
+  }
+
+  if (value) {
+    return value
+      .split(/,|\n/)
+      .map((entry) => entry.trim())
+      .filter((entry) => !!entry);
+  }
+
+  return [];
+}
+
+module.exports = new Command("appdistribution:distribute <distribution-file>")
+  .description("upload a distribution")
+  .option("--app <app_id>", "the app id of your Firebase app")
+  .option("--release-notes <string>", "release notes to include with this distribution")
+  .option(
+    "--release-notes-file <file>",
+    "path to file with release notes to include with this distribution"
+  )
+  .option("--testers <string>", "a comma separated list of tester emails to distribute to")
+  .option(
+    "--testers-file <file>",
+    "path to file with a comma separated list of tester emails to distribute to"
+  )
+  .option("--groups <string>", "a comma separated list of group aliases to distribute to")
+  .option(
+    "--groups-file <file>",
+    "path to file with a comma separated list of group aliases to distribute to"
+  )
+  .before(requireAuth)
+  .action(async (file: string, options: any) => {
+    const appId = getAppId(options.app);
+    const distribution = getDistribution(file);
+    const releaseNotes = getReleaseNotes(options.releaseNotes, options.releaseNotesFile);
+    const testers = getTestersOrGroups(options.testers, options.testersFile);
+    const groups = getTestersOrGroups(options.groups, options.groupsFile);
+    const requests = new req.AppDistributionRequests(appId);
+
+    try {
+      await requests.provisionApp();
+    } catch (err) {
+      throw new FirebaseError(`failed to provision app with ${err.message}`, { exit: 1 });
+    }
+
+    const releaseHash = await new Promise<string>((resolve) => {
```

Comment:
Why is this part of the command and not just a method like `releaseHash()` on `Distribution`?

Reply:
Great idea, done.